src/bsoncxx/document/value.cpp

```diff
@@ -23,6 +23,9 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace document {
 
+value::value() noexcept : _data(NULL, [](std::uint8_t*) {}) {
```

Comment:
so this is a little hacky... the alternative is for the view_or_value class to hold an optional of value instead, then we can drop the requirement on value types being default constructible.